soc/silabs/silabs_siwx91x/siwg917/nwp.c

```diff
@@ -20,14 +20,22 @@
 #include "rsi_ble_common_config.h"
 #endif
 
+#define AP_MAX_NUM_STA 4
+
 LOG_MODULE_REGISTER(siwx91x_nwp);
 
 BUILD_ASSERT(DT_REG_SIZE(DT_CHOSEN(zephyr_sram)) == KB(196) ||
 	     DT_REG_SIZE(DT_CHOSEN(zephyr_sram)) == KB(256) ||
 	     DT_REG_SIZE(DT_CHOSEN(zephyr_sram)) == KB(320));
 
-int siwx91x_get_nwp_config(sl_wifi_device_configuration_t *get_config, uint8_t wifi_oper_mode)
+int siwx91x_get_nwp_config(sl_wifi_device_configuration_t *get_config, uint8_t wifi_oper_mode,
+			   bool hidden_ssid, uint8_t max_num_sta)
 {
+	if (wifi_oper_mode == WIFI_SOFTAP_MODE && max_num_sta > AP_MAX_NUM_STA) {
+		LOG_ERR("Exceeded maximum supported stations (%d)", AP_MAX_NUM_STA);
+		return -EINVAL;
+	}
```

Comment:
No statement before variable declarations please.

Reply:
I expect to catch the error (with an assert?) if `(hidden_ssid == true || max_num_sta != 0) && wifi_oper_mode != WIFI_SOFTAP_MODE`.